ipe/websockets.go

```diff
@@ -123,13 +123,13 @@ func onMessage(conn *websocket.Conn, w http.ResponseWriter, r *http.Request, ses
 				break
 			}
 
-			isPresence := strings.HasPrefix(channelName, "presence-")
-			isPrivate := strings.HasPrefix(channelName, "private-")
+			isPresence := utils.IsPresenceChannel(channelName)
+			isPrivate := utils.IsPrivateChannel(channelName)
 
 			if isPresence || isPrivate {
 				toSign := []string{connection.SocketID, channelName}
 
-				if isPresence {
+				if isPresence || len(subscribeEvent.Data.ChannelData) > 0 {
```

Comment:
When authenticating a private channel, it is possible to send some payload containing user info, just like presence channels. Thus, if some payload is provided during private channel's authentication, this payload (ChannelData) must be used to generate the auth key.